src/Symfony/Component/HttpFoundation/RequestStack.php

```diff
@@ -100,4 +102,16 @@ public function getParentRequest()
 
         return $this->requests[$pos];
     }
+
+    /**
+     * Gets the current session.
+     */
+    public function getSession(): ?SessionInterface
```

Comment:
are we sure we want to make this nullable as it's meant to be the new way of fetching the session. it would need a null check every time.
when there is no session object like in cli, throwing an exception seems better. and in web context I don't see when there wouldn't be a session ($request->hasSession() return false)?
session creation is done lazy now - only on demand when written. so there shouldn't be a case when there is no session in web context or?

Reply:
well I think you can disable sessions in the famework config which I do for apis. but also then it wouldn't it be better to throw an exception here?

Reply:
and if it's nullable we should clarify in the phpdoc when it might return null so people don't have the same questions as me.

Reply:
Good idea :+1: This make sense. 

I personally don't like calling the `hasSession` method in Request and lot of people don't understand what it means.

Reply:
done